era/base.js

```diff
@@ -107,16 +160,32 @@ export const threatFunctions = {
     extraCoeff = 1
   ) {
     // extraCoeff is only used for tooltip text
-    let a = fight.eventToUnit(ev, "source");
-    let b = fight.eventToUnit(ev, "target");
-    if (!a || !b) return;
-    let coeff = (useThreatCoeffs ? a.threatCoeff(ev.ability) : 1) * extraCoeff;
-    b.addThreat(a.key, amount, ev.timestamp, ev.ability.name, coeff);
+    let source = fight.eventToUnit(ev, "source");
+    let target = fight.eventToUnit(ev, "target");
+    if (!source || !target) return;
+    let coeff = applyThreatCoefficient(
+      useThreatCoeffs ? source.threatCoeff(ev.ability) : BASE_COEFFICIENT,
+      extraCoeff,
+      ev.ability.name
+    );
+    target.addThreat(source.key, amount, ev.timestamp, ev.ability.name, coeff);
```

Comment:
The main change is we pass a structure around instead of  a single coefficient value:

```ts
{ 
  value: 1.16, 
  debug: [
    {value: 1.3, 'Defensive stance'},
    {value: 1.15, 'Defiance (talent)'}
    // ...
  ]
}
```